server/src/main/java/org/opensearch/tasks/consumer/TopNSearchTasksLogger.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tasks.consumer;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.search.SearchShardTask;
+import org.opensearch.common.collect.Tuple;
+import org.opensearch.common.logging.Loggers;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.tasks.ResourceStats;
+import org.opensearch.tasks.Task;
+
+import java.util.Comparator;
+import java.util.PriorityQueue;
+import java.util.Queue;
+import java.util.function.Consumer;
+
+/**
+ * A simple listener that logs resource information of high memory consuming search tasks
+ */
+public class TopNSearchTasksLogger implements Consumer<Task> {
+    public static final String TASK_DETAILS_LOG_PREFIX = "task.detailslog";
+    public static final String LOG_TOP_QUERIES_SIZE = "cluster.task.consumers.topn.size";
```

Comment:
nit: separate out topn as `top_n`
1.  While logging should there be a minimum threshold for resource consumption
2. Although the comment mentions `memory expensive search tasks` I am assuming we could do CPU as well?

Reply:
Modified to top_n. 

I agree that we should introduce a minimum threshold to prevent wasteful logging. Will take this up as an optimization in the next PR.

Reply:
Thanks @sruti1312 please create a tracking issue for the same.